hermes/bin/src/pool.rs

```diff
@@ -0,0 +1,97 @@
+//! Global thread pool for parallel WASM module execution.
+//!
+//! This module initializes a global Rayon thread pool
+//! for running WASM modules in parallel. The pool uses
+//! work-stealing for efficient load balancing across
+//! available threads. To avoid saturating the system,
+//! the pool leaves a small margin of CPU capacity
+//! unused — typically 2 threads are reserved for the
+//! main application flow and event loop.
+//!
+//! Each event may execute multiple WASM modules
+//! concurrently within this pool.
+
+use std::{
+    sync::{atomic::AtomicUsize, Condvar, Mutex},
+    thread::available_parallelism,
+};
+
+use anyhow::{Context, Result};
+use rayon::ThreadPoolBuilder;
+
+/// Global counter of currently running tasks.
+static TASK_COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+/// Synchronization primitives for waiting until all tasks finish.
+static TASK_WAIT: (Mutex<()>, Condvar) = (Mutex::new(()), Condvar::new());
```

Comment:
I feel if the aim here is to build a countdown latch, then we should have a countdown latch type which does that.
Seems like this is the sort of thing we would use in other places.
It would also be easier to test, in isolation.
would also make the intent of it here more obvious.